R/datatypes-classes.R

```diff
@@ -173,6 +173,16 @@ datatype__is_date <- function() {
   self$`_dt`$is_date()
 }
 
+datatype__is_date <- function() {
+  self$`_dt`$is_date()
+}
+
+# TODO-REWRITE: do we have an equivalent of dtype == pl$Datetime, no matter the
+# unit / timezone?
+datatype__is_datetime <- function() {
+  !is.null(self$`_dt`$`_get_datatype_fields`()[["time_unit"]])
+}
```

Comment:
I think this is wrong because Duration also has time_unit.
If we really want to support something like this, we probably need to implement something else on the Rust side.

I am imagining a method that returns sub types.

```r
dtype$`_dt`$`_dtype_name()` # c("polars_dtype_datetime", "polars_dtype_temporal")
class(dtype) # c("polars_data_type", "polars_dtype_datetime", "polars_dtype_temporal", "polars_object")
```

Reply:
> I am imagining a method that returns sub types.

Done in 2f3357d5887653c09b7aed10c544328961ac27e4